types.go

```diff
@@ -278,19 +281,35 @@ func (source *Source) Metadata() []MetadataField {
 
 func (source *Source) AuthenticateToECR() bool {
 	logrus.Warnln("ECR integration is experimental and untested")
-	mySession := session.Must(session.NewSession(&aws.Config{
-		Region:      aws.String(source.AwsRegion),
-		Credentials: credentials.NewStaticCredentials(source.AwsAccessKeyId, source.AwsSecretAccessKey, source.AwsSessionToken),
-	}))
 
-	var config aws.Config
+	var mySession *session.Session
+
+	if source.AwsEC2Credentials {
```

Comment:
Just to make this explicit and consistent with the rest of the file.

```suggestion
	if source.AwsEC2Credentials != "" {
```

Reply:
source.AwsEC2Credentials is a boolean, not string.